odoo/fields.py

```diff
@@ -2297,6 +2297,9 @@ def convert_to_column(self, value, record, values=None, validate=True):
         return super(Selection, self).convert_to_column(value, record, values, validate)
 
     def convert_to_cache(self, value, record, validate=True):
+        if value:
+            value = SelectionStr(value)
+            value.init_vals(self.get_values(record.env), self.name)
```

Comment:
This will be a perf killer. I like the idea of detecting typos, but not by paying a permanent runtime price as a trade-off. There must be another option that is lighter, or only impacts tests.

Also keep in mind that those will only catch typos that should have been detected by reviews and regular unit tests (those selection values must serve some purpose!) - it cannot justify a runtime cost.

Reply:
Maybe we could have a server variable saying if we should run those tests?
Something like
```python
from odoo import DEBUG_CHECKS
if value and DEBUG_CHECK:`
    # instantiate SelectionStr(value)
```

I had another idea similar with the translatable strings with placeholders. I'm sure it is applicable to other things.
While ideally this should be done by a linter, the complexity/modularity of Odoo modules make it almost impossible to do correctly, so doing it at runtime (in debug mode) seems like a good idea to me.
We already do the same kind of checks with Owl when we are in `debug=assets`.

> Also keep in mind that those will only catch typos that should have been detected by reviews and regular unit tests 
Indeed, but I have already seen it/done it multiple times, it is no secret that reviewers can't see everything.

Reply:
I put it in a `test_*` module, where other runtime code checks could be added.
It will make runbot slightly slower, but not production databases.